release/models/network-instance/openconfig-network-instance-l2.yang

```diff
@@ -376,4 +383,308 @@ submodule openconfig-network-instance-l2 {
+          }
+        }
+      }
+    }
+  }
+
+  grouping l2ni-l2rib-mac-table-state {
+    description "L2RIB MAC Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+  }
+
+  grouping l2ni-l2rib-mac-ip-table-state {
+    description "L2RIB Mac-IP Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+
+    leaf host-ip {
+      type inet:ip-address;
+      description "Host IP address for the L2RIB MAC-IP entry";
+    }
+
+    leaf l3-vni {
+      type oc-evpn-types:evi-id;
+      description "Layer3 VNI used for inter-subnet routing";
+    }
+  }
+
+  grouping l2ni-l2rib-common-state {
+    description "L2RIB Common Property Oper Data Grouping";
+
+    leaf mac-address {
+        type yang:mac-address;
+        description "MAC address of the L2RIB MAC or MAC-IP entry";
+    }
+    leaf vlan {
+      type leafref {
+        path "../../../../../../../vlans/vlan/config/vlan-id";
+      }
+      description
+        "VLAN on which the MAC or MAC-IP address is present.";
+    }
+    leaf evi {
+      type oc-evpn-types:evi-id;
+      description "EVPN Instance Identifier for the MAC or MAC-IP";
+    }
+    leaf l2-vni {
+      type oc-evpn-types:evi-id;
+      description "Layer2 VNI segment mapped to given vlan-id";
+    }
+  }
+
+  grouping l2ni-l2rib-common-producer-state {
+    description "L2RIB Common Producer Attributes Oper Data Grouping";
+
+    leaf producer {
+      type enumeration {
+        enum LOCAL {
+          description "local producer source";
+        }
+        enum STATIC {
+          description "static producer source";
+        }
+        enum BGP {
+          description "bgp producer source";
+        }
+      }
+      description "Source of the learned L2RIB route";
+    }
+
+    leaf seq-number {
+      type uint32;
+      description
+        "The sequence number is used to ensure that PEs retain the correct
+         MAC/IP Advertisement route when multiple updates occur for the same
+         MAC address";
+      reference "RFC7432";
+    }
+
+    leaf mobility-state {
+      type enumeration {
+        enum FROZEN {
+          description
+            "Permanently frozen mac-address";
+        }
+        enum DUPLICATE {
+          description
+            "Duplicate mac-address";
+        }
+      }
+      description
+        "Indicates if learned MAC address is duplicate or frozen";
+    }
+
+    leaf esi {
+      type oc-evpn-types:esi;
+      description "Ethernet Segment Identifier";
+    }
+
+    leaf sticky {
+      type boolean;
+      description "MAC address is sticky and not subjected to MAC moves";
+      reference "RFC7432";
+    }
+
+    leaf next-hop {
+      type leafref {
+        path "../../../../next-hops/next-hop/index";
+      }
+        description "Leafref host-ip for the MAC-IP table entry";
+    }
+  }
+
+  grouping l2ni-l2rib-common-next-hop-state {
+    description "L2RIB Common Next Hop Attributes Oper Data Grouping";
+
+    container next-hops {
```

Comment:
Just a quick note/question about this `next-hops` container.  We think that it would be better to move this container to the same level as the `entries` container for both `mac-table` and `mac-ip-table` entries so that entries can share next-hops information. The other approach would be to move the next hop properties directly into the `state` container under entries.

Reply:
+1 to moving to the entries container level.